src/main/scala/tech/cryptonomic/conseil/util/DatabaseUtil.scala

```diff
@@ -144,8 +144,12 @@ object DatabaseUtil {
       * @param ordering list of ordering parameters
       * @return SQLAction with ordering
       */
-    def makeOrdering(ordering: List[QueryOrdering]): SQLActionBuilder = {
-      val orderingBy = ordering.map(x => s"${x.field} ${x.direction}").mkString(",")
+    def makeOrdering(ordering: List[QueryOrdering], aggregation: Option[Aggregation]): SQLActionBuilder = {
+      val aggregatedOrdering = ordering.map {
+        case ord if aggregation.map(_.field).contains(ord.field) => ord.copy(field = mapAggregationToSQL(aggregation.get.function, aggregation.get.field))
```

Comment:
Can do `aggregation.exists(_.field == ord.field)` if you like

Reply:
I might propose 
```scala
val orderingBy = ordering.map {
  ord =>
    val ordField =
      if (aggregation.exists(_.field == ord.field)) mapAggreagationToSQL(..., ...)
      else ord.field
    s"$ordField ${ord.direction}"
}.mkString(",")
```

IMO, this kind of approach reduces the need to "enter/exit" within Lists/Options content using multiple mappings.
It gets to the exact data needed and produces the end result where it's available, so that no extra transformation is done in the calling context if not needed.

Reply:
changed according to your suggestion